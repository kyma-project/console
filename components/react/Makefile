ci-pr: resolve build validate test
ci-master: resolve build validate test
ci-release: resolve build validate test

resolve:
	npm install

build:
	npm run build

validate:
	npm run lint-check
	eslint -c .eslinterrc ./src

test:
	CI=true npm run test
	