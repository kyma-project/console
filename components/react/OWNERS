reviewers:
    - default-reviewers

approvers:
    - default-approvers
    - Wawrzyn321
    - content-reviewers