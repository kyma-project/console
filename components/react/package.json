{
  "name": "@kyma-project/react-components",
  "version": "0.0.43",
  "main": "dist/index.js",
  "private": false,
  "license": "Apache-2.0",
  "directories": {
    "dist": "src"
  },
  "scripts": {
    "start:kyma": "npm run build:watch",
    "start:kyma:api": "npm run build:watch",
    "build": "webpack --config webpack.config.prod.js",
    "build:watch": "webpack -w --config webpack.config.dev.js",
    "test": "jest src/**; exit 0",
    "test:watch": "jest src/** --watch",
    "test:watchAll": "jest src/** --watchAll",
    "test:coverage:report": "open ./coverage/lcov-report/index.html",
    "lint-fix": "prettier --config ../../.prettierrc --write \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\"",
    "lint-check": "prettier --config ../../.prettierrc --check \"./src/**/*.{js,jsx,ts,tsx,html,css,json}\""
  },
  "dependencies": {
    "@babel/runtime-corejs2": "^7.12.5",
    "lodash": "^4.17.21",
    "polished": "^1.8.1",
    "styled-components-modifiers": "^1.0.1"
  },
  "devDependencies": {
    "@babel/types": "7.0.0-beta.44",
    "ajv": "^6.12.3",
    "babel-cli": "^6.26.0",
    "babel-core": "^6.26.0",
    "babel-eslint": "^8.0.1",
    "babel-jest": "^22.4.1",
    "babel-loader": "^7.1.2",
    "babel-polyfill": "^6.26.0",
    "babel-preset-env": "^1.6.1",
    "babel-preset-react": "^6.24.1",
    "babel-preset-stage-0": "^6.24.1",
    "enzyme": "^3.1.0",
    "enzyme-adapter-react-16": "^1.0.2",
    "exec-sh": "^0.3.3",
    "file-loader": "^1.1.11",
    "fiori-fundamentals": "1.5.3",
    "fontfaceobserver": "^2.0.13",
    "handlebars": "^4.7.3",
    "jest": "^24.0.0",
    "jest-styled-components": "^4.11.0-1",
    "npm-run-all": "^4.1.1",
    "raf": "^3.4.0",
    "react-jsonschema-form": "1.7.0",
    "react-modal": "^3.5.1",
    "react-scrollbar": "^0.5.4",
    "react-test-renderer": "^16.0.0",
    "style-loader": "^0.20.2",
    "styled-components": "^3.2.3",
    "styled-components-grid": "^2.2.0",
    "webpack": "^4.28.1",
    "webpack-cli": "^3.2.1",
    "webpack-merge": "^4.2.1"
  },
  "peerDependencies": {
    "styled-components": "^3.2.3"
  },
  "resolutions": {
    "uglify-es": "3.3.7"
  },
  "jest": {
    "collectCoverage": true,
    "collectCoverageFrom": [
      "src/**",
      "!src/**/*.snap",
      "!src/components/index.js",
      "!src/modifiers/**",
      "!src/index.js"
    ],
    "moduleNameMapper": {
      "^.+\\.(css|less|scss)$": "babel-jest"
    },
    "moduleDirectories": [
      "node_modules",
      "./"
    ],
    "setupFiles": [
      "raf/polyfill",
      "./test_setup/config"
    ]
  }
}
