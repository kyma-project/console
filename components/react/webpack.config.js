const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');

require('babel-polyfill');

let libraryName = pkg.name;

module.exports = {
  entry: ['./src/index'],
  module: {
    loaders: [
      { test: /\.js?$/, loader: 'babel-loader', exclude: /node_modules/ },
      {
        test: /\.css$/,
        use: [{ loader: 'style-loader' }, { loader: 'css-loader' }],
      },
      {
        test: /\.(eot|svg|ttf|woff|woff2|otf)$/,
        loader: 'file-loader?name=fonts/[name].[ext]',
      },
    ],
  },
  resolve: {
    modules: [path.resolve('./node_modules'), 'node_modules'],
    extensions: ['.js'],
  },
  output: {
    path: path.join(__dirname, '/lib'),
    publicPath: '/',
    filename: 'index.js',
    library: libraryName,
    libraryTarget: 'umd',
    umdNamedDefine: true,
  },
  devtool: 'source-map',
  devServer: {
    contentBase: './src',
    hot: true,
  },
  plugins: [
    new webpack.optimize.OccurrenceOrderPlugin(),
    new webpack.HotModuleReplacementPlugin(),
    new webpack.NoEmitOnErrorsPlugin(),
  ],
  externals: {
    'styled-components': {
      commonjs: 'styled-components',
      commonjs2: 'styled-components',
      amd: 'styled-components',
    },
  },
};
