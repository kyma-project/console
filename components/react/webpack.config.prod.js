const webpack = require('webpack');
const path = require('path');
const pkg = require('./package.json');
const webpackConfigBase = require('./webpack.config');
require('babel-polyfill');

const plugins = Array.from(webpackConfigBase.plugins);

plugins.push(
  new webpack.DefinePlugin({
    'process.env.NODE_ENV': JSON.stringify('production'),
  }),
);

const webpackConfig = {
  ...webpackConfigBase,
  devtool: false,
  plugins,
};

module.exports = webpackConfig;
