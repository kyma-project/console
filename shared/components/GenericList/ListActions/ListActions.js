import React from 'react';
import { Popover, Menu, Button } from 'fundamental-react';

const AUTO_ICONS_BY_NAME = new Map([['Edit', 'edit'], ['Delete', 'delete']]);

const StandaloneAction = ({ action }) => {
  const icon = action.icon || AUTO_ICONS_BY_NAME.get(action.name);

  return (
    <Button
      onClick={entry => action.handler(entry)}
      className="generic-list-actions__standalone"
      option="light"
      glyph={icon}
    >
      {icon ? '' : action.name}
    </Button>
  );
};

const ListActions = ({ actions, entry, standaloneItems = 2 }) => {
  if (!actions.length) {
    return null;
  }

  const listItems = actions.slice(standaloneItems, actions.length);

  return (
    <div className="generic-list-actions">
      {actions.slice(0, standaloneItems).map(a => (
        <StandaloneAction key={a.name} action={a} entry={entry} />
      ))}
      {listItems.length ? (
        <Popover
          body={
            <Menu>
              <Menu.List>
                {listItems.map(a => (
                  <Menu.Item onClick={() => a.handler(entry)} key={a.name}>
                    {a.name}
                  </Menu.Item>
                ))}
              </Menu.List>
            </Menu>
          }
          control={<Button glyph="vertical-grip" option="light" />}
          placement="bottom-end"
        />
      ) : null}
    </div>
  );
};

export default ListActions;
