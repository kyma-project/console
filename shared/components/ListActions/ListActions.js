import React from 'react';
import { Popover, Menu, Button } from 'fundamental-react';
import './ListActions.scss';
import PropTypes from 'prop-types';

const AUTO_ICONS_BY_NAME = new Map([['Edit', 'edit'], ['Delete', 'delete']]);

const StandaloneAction = ({ action, entry }) => {
  const icon = action.icon || AUTO_ICONS_BY_NAME.get(action.name);

  return (
    <Button
      onClick={() => action.handler(entry)}
      className="list-actions__standalone"
      option="light"
      glyph={icon}
      aria-label={action.name}
    >
      {icon ? '' : action.name}
    </Button>
  );
};

const ListActions = ({ actions, entry, standaloneItems = 2 }) => {
  if (!actions.length) {
    return null;
  }

  const listItems = actions.slice(standaloneItems, actions.length);

  return (
    <div className="list-actions">
      {actions.slice(0, standaloneItems).map(a => (
        <StandaloneAction key={a.name} action={a} entry={entry} />
      ))}
      {listItems.length ? (
        <Popover
          body={
            <Menu>
              <Menu.List>
                {listItems.map(a => (
                  <Menu.Item onClick={() => a.handler(entry)} key={a.name}>
                    {a.name}
                  </Menu.Item>
                ))}
              </Menu.List>
            </Menu>
          }
          control={
            <Button
              glyph="vertical-grip"
              option="light"
              aria-label="more-actions"
            />
          }
          placement="bottom-end"
        />
      ) : null}
    </div>
  );
};

ListActions.propTypes = {
  actions: PropTypes.arrayOf(
    PropTypes.shape({
      name: PropTypes.string,
      handler: PropTypes.func.isRequired,
    }),
  ).isRequired,
  entry: PropTypes.any.isRequired,
  standaloneItems: PropTypes.number,
};

export default ListActions;
