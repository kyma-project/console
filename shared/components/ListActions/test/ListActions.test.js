import React from 'react';
import ListActions from '../ListActions';

import 'core-js/es/array/flat-map';
import { render, fireEvent } from '@testing-library/react';

describe('ListActions', () => {
  // it("Skips rendering actions when 'actions' prop passes skipAction() call", () => {
  //   let actions = [
  //     { name: 'testaction', handler: () => {}, skipAction: () => true },
  //   ];
  //   let component = renderer.create(
  //     <GenericList
  //       title=""
  //       actions={actions}
  //       entries={mockEntries}
  //       headerRenderer={mockHeaderRenderer}
  //       rowRenderer={mockEntryRenderer}
  //     />,
  //   );
  //   let tree = component.toJSON();
  //   expect(tree).toMatchSnapshot();
  // });
});
