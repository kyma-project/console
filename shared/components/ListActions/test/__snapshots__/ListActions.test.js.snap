// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListActions Renders only dropdown actions 1`] = `
Array [
  Array [
    "Warning: Each child in a list should have a unique \\"key\\" prop. See https://fb.me/react-warning-keys for more information.%s",
    "
    in Menu.Item (created by ListActions)
    in ul (created by Menu.List)
    in Menu.List (created by ListActions)
    in nav (created by Menu)
    in Menu (created by ListActions)
    in div (created by InnerPopper)
    in InnerPopper (created by Context.Consumer)
    in Popper
    in div (created by Foco)
    in Foco
    in Manager
    in Popper
    in div (created by Popover)
    in Popover (created by ListActions)
    in div (created by ListActions)
    in ListActions",
  ],
]
`;
